### PR TITLE
Add Tutorial json storage

### LINKED ISSUE #92: Add Tutorial to json storage

### PR DESCRIPTION
**Pull request by BlazeChron**
Closes #92 and #96 

Adds tutorial field to addressbook json. Extra class created to support this.
Adjusted relevant cases to support new field.

### CODE REVIEW

File: src/main/java/seedu/address/model/tutorial/UniqueTutorialList.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.tutorial.exceptions.DuplicateTutorialException;
+import seedu.address.model.tutorial.exceptions.TutorialNotFoundException;
```
Comment (gladwintan) [inline src/main/java/seedu/address/model/tutorial/UniqueTutorialList.java:12]: Should there be a newline here after the imports

File: src/main/java/seedu/address/model/tutorial/UniqueTutorialList.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+     * The tutorial must exist in the list.
+     */
+    public void remove(Tutorial toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new TutorialNotFoundException();
+        }
+    }
+
+
```
Comment (gladwintan) [inline src/main/java/seedu/address/model/tutorial/UniqueTutorialList.java:81]: extra line

File: src/main/java/seedu/address/storage/JsonAdaptedTutorial.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.tutorial.Tutorial;
+
+
```
Comment (gladwintan) [inline src/main/java/seedu/address/storage/JsonAdaptedTutorial.java:9]: extra line

**Review by gladwintan — APPROVED** [review]
nit

**Review by jinhanjinhan — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/tutorial/UniqueTutorialList.java (hunk 1 above)
Comment (BlazeChron) [reply src/main/java/seedu/address/model/tutorial/UniqueTutorialList.java:12]: image
The checkstyle is failing if I add a line separation, perhaps not?

**Pull request merged**